src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -574,6 +586,20 @@ internal int ExecuteTool(string pathToTool, string responseFileCommands, string
             }
 
             return ExitCode;
+
+            // Construct the friendly name for the compilation. This does not need to be unique. Instead
+            // it's used by developers to understand what compilation is running on the server.
+            string GetRequestId()
+            {
+                if (!string.IsNullOrEmpty(ProjectName))
+                {
+                    return string.IsNullOrEmpty(TargetFramework)
+                        ? ProjectName
+                        : $"{ProjectName} ({TargetFramework})";
+                }
+
+                return $"Unnamed compilation";
```

Comment:
```suggestion
                return "Unnamed compilation";
```

Reply:
Had to add `$` when I put the GUID back in.